pkg/infradriver/ssh_infradriver.go

```diff
@@ -117,7 +118,36 @@ func NewInfraDriver(cluster *v2.Cluster) (InfraDriver, error) {
 		return nil, fmt.Errorf("no hosts specified")
 	}
 
-	if err := checkAllHostsSameFamily(ret.hosts); err != nil {
+	// check registry config is valid,
+	// make sure external registry domain is valid
+	if cluster.Spec.Registry.ExternalRegistry != nil {
```

Comment:
in the future, we should not distinguish the local registry and external registry

Reply:
add TODO here